### ISSUE TITLE
Do Mess roms work?

### ISSUE THREAD

**Issue by ecstaticax** [yoshisuga/MAME4iOS#348]
Hi, 
I tried some computer roms, but they don't work. Are they supported? Where can I find a working  computer romset (if it is possible to ask for it)?

**Comment by eduo** [comment]
I'm not sure they're supported but if they are, they might require you to use .7z instead of .zip, if internally that's what's selecting the more modern MAME rather than the older one.

**Comment by ToddLa** [comment]
I have not tested every `MESS`, but yes they work, if you install them correctly. 

Basically look at the `HELP` or `README` under `How to add software to MAME4iOS`

The gist is that you need to do three things...
1. Import the Software List XML
2. Import any BIOS/MACHINE romset. Some systems are "romless" and don't need this step. 
3. Import the game romsets you want to play.

**Comment by ToddLa** [comment]
> I'm not sure they're supported but if they are, they might require you to use .7z instead of .zip, if internally that's what's selecting the more modern MAME rather than the older one.

MAME handles `zip` and `7z` exactly archives the same, there is no advantage to one over the other.

...except for minor file size improvements you get with `7z`

**Comment by eduo** [comment]
I'm sorry. I didn't mean to say there're advantages of 7z over zip. I personally don't even like the format.

What I meant is that the newer version specifically handles 7z files with  0.234 but the format is ignored by the older 0.139, as stated in the release.

**Comment by eduo** [comment]
> I have not tested every `MESS`, but yes they work, if you install them correctly.
> 
> Basically look at the `HELP` or `README` under `How to add software to MAME4iOS`
> 
> The gist is that you need to do three things...
> 
> 1. Import the Software List XML
> 2. Import any BIOS/MACHINE romset. Some systems are "romless" and don't need this step.
> 3. Import the game romsets you want to play.

This ticket is complemente by #342 as it makes it clear step 1 is no longer needed in newer releases and even lists which systems are romless for step 2. 

I haven't tried this yet. I will probably try with the Spectrum machine and the Game & Watch (not sure if keyboard layouts and the G&W artwork needs to be added as well, I have never tried it) in the iPad and then tvOS.

**Comment by ToddLa** [comment]
Yes as of 2021.8 all MAME software list XMLs are "built-in" no need to install an XML.

**Comment by ToddLa** [comment]
...and as of 2022.1 you can install and run standalone roms, you don't need to use the SoftwareList system.

**Comment by ToddLa** [comment]
I have tried GnW romsets and they do work, i even found and fixed a rendering bug that happens with GnW artwork!

If you are running tvOS 14+ a mouse and keyboard connected to the TV should work just fine....we don't do any emulation of keyboard or mouse on tvOS (unlike iOS) you must have hardware devices.

**Comment by eduo** [comment]
> I have tried GnW romsets and they do work, i even found and fixed a rendering bug that happens with GnW artwork!
> 
> If you are running tvOS 14+ a mouse and keyboard connected to the TV should work just fine....we don't do any emulation of keyboard or mouse on tvOS (unlike iOS) you must have hardware devices.

I understand you mean paired keyboard and mouse. This is of course perfect. :D
